src/util/debug.hpp

```diff
@@ -11,6 +11,13 @@ namespace nest {
 namespace mc {
 namespace util {
 
+constexpr inline bool is_debug_mode() {
+#ifndef NDEBUG
+    return true;
+#else
+    return false;
+#endif
+}
```

Comment:
For another PR: we are going to have to think carefully about run-time checking; it should be independent of what CMake wants to do with Release vs. Debug default flags.